### PR TITLE
add spec for x/derivatives

### PR DESCRIPTION
**Pull request by taiki1frsh**
fill spec

### CODE REVIEW

**Comment by taiki1frsh** [comment]
@KimuraYu45z @mkXultra 

I wrote down the fundamental features and specifications in a text besides basic data structures.
Please review and teach me if there's a lacking part or a wrong description, or so.

File: x/derivatives/spec/01_concepts.md [hunk 1]
```diff
@@ -1,15 +1,73 @@
...
+Our liquidity provider token's ticker is `DLP`.   
+In the backend, it has `udlp` denom, which is the micro unit of `DLP`.   
+
+DLP consists of an index of assets used for leverage trading. It can be minted using the assets which the protocol supports and burnt to redeem any index asset. The price for minting and redemption is calculated based on the formulas in the WhitePaper in the section "3.1.1".   
+WhitePaper: https://ununifi.io/assets/download/UnUniFi-Whitepaper.pdf
+
+Fees earned on the platform are directly added to the pool.　Therefore, DLP holders can benefit from them as a reward through the increasement of the DLP price.   
 
-## Liquidity Provider Token
+There's no dynamic change of the minting and redemption fee rate at this moment unlike GMX. It's the static rate which is defined in the protocol. But, it can be modified through the governace voting.
```
Comment (KimuraYu45z) [inline x/derivatives/spec/01_concepts.md:21]: This is unintended.

https://github.com/UnUniFi/chain/blob/newDevelop/x/derivatives/keeper/lpt.go#L81

File: x/derivatives/spec/01_concepts.md [hunk 2]
```diff
@@ -1,15 +1,73 @@
...
 ## Perpetual Futures
 
-- Imaginary Funding Rates are levied at every 8 hours.
+### Position
+
+User can open a perpetual futures position by sending `MsgOpenPosition` tx. There's no fee for opening a position.   
+The position can be covered by two types of asset as margin, which are the tokens of the trading pair. If you trade 'BTC/USDC' pair, you can deposit BTC or USDC as margin. The profit will be distributed in the same token as the margin if there's some.    
+The created position cannot be modified except for closing a whole in the current implementation.    
+And, the liquidation is triggered against each position. The margin of the position cannot be added afterward now. But, this will be supported in the near future.   
+The max leverate rate is defined in the params  of the protocol for all trading pairs equially. This can be modified through the governance voting.
```
Comment (KimuraYu45z) [inline x/derivatives/spec/01_concepts.md:34]: `leverate`
`leverage`

**Review by KimuraYu45z — CHANGES_REQUESTED** [review]

File: x/derivatives/spec/01_concepts.md (hunk 1 above)
Comment (taiki1frsh) [reply x/derivatives/spec/01_concepts.md]: I modified it.
Please check the above line.

File: x/derivatives/spec/01_concepts.md (hunk 2 above)
Comment (taiki1frsh) [reply x/derivatives/spec/01_concepts.md]: I fixed them.

File: x/derivatives/spec/01_concepts.md [hunk 3]
```diff
@@ -1,15 +1,73 @@
...
+Our liquidity provider token's ticker is `DLP`.   
+In the backend, it has `udlp` denom, which is the micro unit of `DLP`.   
+
+DLP consists of an index of assets used for leverage trading. It can be minted using the assets which the protocol supports and burnt to redeem any index asset. The price for minting and redemption is calculated based on the formulas in the WhitePaper in the section "3.1.1".   
+WhitePaper: https://ununifi.io/assets/download/UnUniFi-Whitepaper.pdf
+
+Fees earned on the platform are directly added to the pool.　Therefore, DLP holders can benefit from them as a reward through the increasement of the DLP price.   
 
-## Liquidity Provider Token
+There's dynamic change of the minting and redemption fee rate at this moment. There's the static rate which is defined in the protocol. And, the actual fee rate also consider the difference of asset proportion between target and actual proportion.  The static base fee rate can be modified through the governace voting.
```
Comment (KimuraYu45z) [inline x/derivatives/spec/01_concepts.md:22]: Plz add something like

```
mintFeeRate is proportion to max(0, (actualAmountInPool[i] - targetAmount[i]) / targetAmount[i])
redeemFeeRate is proportion to max(0, -(actualAmountInPool[i] - targetAmount[i]) / targetAmount[i])
```

**Review by KimuraYu45z — APPROVED** [review]
Just one remaining thing should be added, but others LGTM.

File: x/derivatives/spec/01_concepts.md (hunk 3 above)
Comment (taiki1frsh) [reply x/derivatives/spec/01_concepts.md:22]: Okay, thanks for the content.
Will put it.

By the way, I (and @mkXultra ) are planning to increment those more technical specs hereafter the general spec on text is fixed.

**Pull request merged**